test/blob_test.js

```diff
@@ -332,36 +367,743 @@ suite.only("Azure Blob", function() {
+        blobType: 'BlockBlob'
+      };
+      var content = 'hello world';
+      return blob.putBlob(containerName, blockBlobName, options, content);
+    });
+
+    test('get text block blob', function(){
+      return blob.getBlob(containerName, blockBlobName).then(function(result){
+        assert(result.content === 'hello world');
+      });
+    });
+
+    test('set and get blob metadata', function() {
+      var metadata = {
+        origin: 'taskcluster'
+      };
+      return blob.setBlobMetadata(containerName, blockBlobName, metadata)
+        .then(function() {
+          return blob.getBlobMetadata(containerName, blockBlobName);
+        })
+        .then(function(result){
+          assert(result.metadata.origin === 'taskcluster');
+        });
+    });
+
+    test('set and get blob properties', function() {
+      var options = {
+        cacheControl: 'no-cache',
+        contentType: 'text/plain;charset="utf8"',
+        contentEncoding: 'gzip',
+        contentLanguage: 'en-US',
+        contentDisposition: 'attachment; filename="file.txt"'
+      }
+
+      return blob.setBlobProperties(containerName, blockBlobName, options)
+        .then(function() {
+          return blob.getBlobProperties(containerName, blockBlobName);
+        })
+        .then(function(result) {
+          assert(result.blobType === 'BlockBlob');
+          assert(result.contentLength === '11');
+          assert(result.contentType === 'text/plain;charset="utf8"');
+          assert(result.contentEncoding === 'gzip');
+          assert(result.contentLanguage === 'en-US');
+          assert(result.contentDisposition === 'attachment; filename="file.txt"');
+          assert(result.metadata.origin === 'taskcluster');
+        });
+    });
+
+    test('put block, getBlockList and putBlockList', function () {
+      var blockContent1 = Buffer.from("Lorem Ipsum is simply dummy text of the printing and typesetting industry. " +
+        "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a" +
+        " galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also" +
+        " the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the " +
+        "release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.");
+      var blockId1 = blob.getBlockId('fastazure', 1, 3);
+
+      var blockContent2 = 'Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old.';
+      var blockId2 = blob.getBlockId('fastazure', 2, 3);
+
+      var options = {
+        blockId: blockId1
+      }
+      // put a new block to a block blob
+      return blob.putBlock(containerName, blockBlobName, options, blockContent1)
```

Comment:
So this .then chain gets really hard to read and seems to increase indentation at line 591 without explanation...

I've found a really nice pattern for working with long promise chains without async and await is this:
```js
var p = firstPromise();

p = p.then(() => {});

p = p.then(() => {});

return p;
```
it makes the code easier to read and debug in my opinion.